extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -14,188 +15,197 @@ class CuratedContentValidator {
 	const ERR_EMPTY_LABEL = 'emptyLabel';
 	const ERR_NO_CATEGORY_IN_TAG = 'noCategoryInTag';
 
-	private $errors;
-	private $existingSectionLabels;
-	private $existingItemLabels;
-	private $existingFeaturedItemLabels;
-	private $hasOptionalSection;
-
-	public function __construct() {
-		$this->reset();
-	}
-
-	public function reset() {
-		$this->errors = [ ];
-		$this->existingSectionLabels = [ ];
-		$this->existingItemLabels = [ ];
-		$this->existingFeaturedItemLabels = [ ];
-		$this->hasOptionalSection = false;
-	}
-
 	public function validateData( $data ) {
-		$this->reset();
+		$errors = [ ];
+		$alreadyUsedSectionLabels = [ ];
+
+		// If data is empty or not an array - return empty array.
+		// So users have ability to erase all data by sending empty value.
+		if ( !is_array( $data ) ) {
+			return [];
+		}
 
-		// validate sections
 		foreach ( $data as $section ) {
 			if ( !empty( $section['featured'] ) ) {
-				$this->validateItems( $section, true );
+				if ( !$this->validateItemsInFeatured( $section ) ) {
+					$errors[] = self::ERR_OTHER_ERROR;
+				}
 			} else {
-				$this->validateSection( $section );
-				$this->validateItemsExist( $section );
-				$this->validateItems( $section );
-				$this->validateItemsTypes( $section );
+				$alreadyUsedSectionLabels[] = $section['title'];
+				if ( empty( $section['title'] ) ) {
+					if ( !$this->validateItemsInSection( $section ) ) {
+						$errors[] = self::ERR_OTHER_ERROR;
+					} ;
+				} elseif ( !empty( $this->validateSectionWithItems( $section ) ) ) {
+					$errors[] = self::ERR_OTHER_ERROR;
+				} ;
```

Comment:
This piece of code isn't really self explanatory so adding comments would be nice.

Also what's up with `} ;`?